### PR TITLE
FEAT-#4524: Split Modin API and Memory log files

### LINKED ISSUE #4524: Split logs into multiple separate files
Right now we have different kinds of logs intermixed into a single log file. For starters, I'd like to separate the memory usage history into a separate log.

This will require creating a logs folder and putting all logs there.

### PR DESCRIPTION
**Pull request by naren-ponder**
Signed-off-by: Naren Krishna <naren@ponder.io>

## What do these changes do?

In the current approach to Modin logging, all the logs are logged to a single file. In this updated approach, we separate out the memory logs from the Modin API logs.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4524 ? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/logging/config.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def configure_logging():
     logfile = RotatingFileHandler(
         filename=log_filename,
         mode="a",
-        maxBytes=LogMemorySize.get() * int(1e5),
+        maxBytes=int(1e7),
```
Comment (naren-ponder) [inline modin/logging/config.py:103]: We make the `LogMemorySize` configurable. I hardcoded the API call logs to default to 10MB per file (same default at the memory logs). I do not think it is necessary to make this configurable as well, but open to other opinions.

File: modin/logging/config.py [hunk 2]
```diff
@@ -120,25 +120,58 @@ def configure_logging():
...
         mem = threading.Thread(
-            target=memory_thread, args=[logger, mem_sleep], daemon=True
+            target=memory_thread, args=[mem_logger, mem_sleep], daemon=True
         )
         mem.start()
 
     __LOGGER_CONFIGURED__ = True
 
 
+def configure_memory_logging(job_id, formatter):
```
Comment (mvashishtha) [inline modin/logging/config.py:139]: This function is instantiating and configuring the logger, but it's also logging the initial memory messages. I think it's doing too much at once. I think it's actually ok to get rid of this function and inline its body in `configure_logging`. If you don't want to do that, I'd prefer calling it `get_memory_logger` and moving the initial log messages into `configure_logging`.

File: modin/logging/config.py [hunk 3]
```diff
@@ -120,25 +120,58 @@ def configure_logging():
...
+    -------
+    logging.Logger
+        The Modin memory logger.
+    """
+    mem_logger = logging.getLogger("modin_memory.logger")
+    mem_log_filename = f".modin/logs/memory_{job_id}.log"
+    logfile = RotatingFileHandler(
+        filename=mem_log_filename,
+        mode="a",
+        maxBytes=LogMemorySize.get() * int(1e5),
```
[the remark this answers is not in the document]
Comment (naren-ponder) [reply modin/logging/config.py:160]: You're right, good catch!

File: modin/logging/config.py [hunk 4]
```diff
@@ -120,19 +120,31 @@ def configure_logging():
...
     logger.info(f"Physical Cores: {num_physical_cores}")
     logger.info(f"Total Cores: {num_total_cores}")
-    logger.info(f"Memory Total: {bytes_int_to_str(svmem.total)}")
-    logger.info(f"Memory Available: {bytes_int_to_str(svmem.available)}")
-    logger.info(f"Memory Used: {bytes_int_to_str(svmem.used)}")
 
     if LogMode.get() != "enable_api_only":
         mem_sleep = LogMemoryInterval.get()
+        mem_logger = logging.getLogger("modin_memory.logger")
+        mem_log_filename = f".modin/logs/memory_{job_id}.log"
```
Comment (devin-petersohn) [inline modin/logging/config.py:131]: Can we structure these logs so they are in a directory?

Comment (naren-ponder) [reply modin/logging/config.py:131]: You mean a directory on a per job basis? Like `.modin/logs/job_<uuid>/api.1` and `.modin/logs/job_<uuid>/memory.1`?

Comment (devin-petersohn) [reply modin/logging/config.py:131]: @naren-ponder yes, except what is the `.1` component?

**Review by mvashishtha — APPROVED** [review]
LGTM! Thank you @naren-ponder

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 5]
```diff
@@ -3,15 +3,20 @@ Modin Logging
...
+The logs generated by Modin Logging will be written to a `.modin/logs/job_<uuid>` directory, uniquely named after the job uuid.
+The logs that contain the Modin API stack traces are named `trace.log`. The logs that contain the memory utilization metrics are 
+named `memory.log`. By default, if any log file exceeds 10MB, the logs will rollover and several files will be created. For instance, if 
+users have 20MB worth of Modin API logs, you can expect to find `trace.log.1` and `trace.log.2` in the `.modin/logs/job_<uuid>` directory.
 
-Usage example
--------------
+**Developer Warning:** In some cases, running services like JupyterLab in the `modin/modin` directory may result in circular dependency issues. 
+This is as a result of a name conflict between the `modin/logging` directory and the Python `logging` module, which may be used as a default in 
+such environments. As a resolution, one one could run Jupyterlab or other services from any directory except `modin/modin`.
```
Comment (YarShev) [inline docs/usage_guide/advanced_usage/modin_logging.rst:16]:
```suggestion
such environments. As a resolution, one could run Jupyterlab or other services from any directory except `modin/modin`. 
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 6]
```diff
@@ -3,15 +3,20 @@ Modin Logging
...
+in addition to Modin API calls (see the usage examples below).
 
-**Developer Warning:** In some cases, running Modin logging inside of a Jupyterlab or other notebook instance at the root level of the Modin 
-repository or in the `modin/modin` directory may result in circular dependency issues. This is as a result of a name conflict between the 
-`modin/logging` directory and the Python `logging` module, which may be used as a default in such environments. As a resolution, one can simply
-run Modin logging from a different directory or manually manage the name conflicts.
+The logs generated by Modin Logging will be written to a `.modin/logs/job_<uuid>` directory, uniquely named after the job uuid.
+The logs that contain the Modin API stack traces are named `trace.log`. The logs that contain the memory utilization metrics are 
+named `memory.log`. By default, if any log file exceeds 10MB, the logs will rollover and several files will be created. For instance, if 
+users have 20MB worth of Modin API logs, you can expect to find `trace.log.1` and `trace.log.2` in the `.modin/logs/job_<uuid>` directory.
```
Comment (RehanSD) [inline docs/usage_guide/advanced_usage/modin_logging.rst:12]:
```suggestion
users have 20MB worth of Modin API logs, they can expect to find `trace.log.1` and `trace.log.2` in the `.modin/logs/job_<uuid>` directory.
```

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 7]
```diff
@@ -3,15 +3,20 @@ Modin Logging
...
 
-Usage example
--------------
+**Developer Warning:** In some cases, running services like JupyterLab in the `modin/modin` directory may result in circular dependency issues. 
+This is as a result of a name conflict between the `modin/logging` directory and the Python `logging` module, which may be used as a default in 
+such environments. As a resolution, one could run Jupyterlab or other services from any directory except `modin/modin`.
```
Comment (RehanSD) [inline docs/usage_guide/advanced_usage/modin_logging.rst:16]:
```suggestion
such environments. To resolve this, please run Jupyterlab or other similar services from directories other than  `modin/modin`. 
```

File: modin/config/envvars.py [hunk 8]
```diff
@@ -415,38 +415,38 @@ def get(cls):
...
 
 
-class LogMemorySize(EnvironmentVariable, type=int):
-    """Size of logs (in MBs) to store per Modin job."""
+class LogFileSize(EnvironmentVariable, type=int):
+    """Max size of logs (in MBs) to store per Modin job."""
 
-    varname = "MODIN_LOG_MEMORY_SIZE"
+    varname = "MODIN_LOG_FILE_SIZE"
     default = 100
```
Comment (RehanSD) [inline modin/config/envvars.py:422]: Doesn't this set the default to 100 MB rather than 10MB as the doc above says?

**Review by RehanSD — CHANGES_REQUESTED** [review]
Looks good to me - left some minor comments. Couple of overall comments - 
* I believe you need to use \`\` when writing in-line code in `.rst` files rather than just \` by itself. (e.g. \`\`code formatted text\`\`)
* The tense of the docs change from referring to a user and telling them what to expect to speaking in first person - would be good to unify that
* I noticed that the file size for both traces and memory logs are the same - is that by design, or have we considered allowing different file sizes for each? I imagine it could be possible for a user to want more of one than the other e.g., but this may also be over engineering.

File: modin/config/envvars.py (hunk 8 above)
Comment (naren-ponder) [reply modin/config/envvars.py:422]: Yes good catch!

**Comment by naren-ponder** [comment]
The memory and API log sizes being the same is mostly a coincidence (was not designed that way), but I think making that configurable is not worth it.

File: docs/usage_guide/advanced_usage/modin_logging.rst [hunk 9]
```diff
@@ -1,39 +1,46 @@
...
-run Modin logging from a different directory or manually manage the name conflicts.
+The logs generated by Modin Logging will be written to a ``.modin/logs/job_<uuid>`` directory, uniquely named after the job uuid.
+The logs that contain the Modin API stack traces are named ``trace.log``. The logs that contain the memory utilization metrics are
+named ``memory.log``. By default, if any log file exceeds 10MB, the logs will rollover and several files will be created. For instance, if
+users have 20MB worth of Modin API logs, they can expect to find ``trace.log.1`` and ``trace.log.2`` in the ``.modin/logs/job_<uuid>`` directory.
 
-Usage example
--------------
+**Developer Warning:** In some cases, running services like JupyterLab in the ``modin/modin`` directory may result in circular dependency issues.
+This is as a result of a name conflict between the ``modin/logging`` directory and the Python ``logging`` module, which may be used as a default in
```
Comment (RehanSD) [inline docs/usage_guide/advanced_usage/modin_logging.rst:15]:
```suggestion
This is due to a naming conflict between the ``modin/logging`` directory and the Python ``logging`` module, which may be used as a default in
```

**Review by RehanSD — CHANGES_REQUESTED** [review]
Left a quick comment. There's a shift in the docs from passive voice in the intro to active voice in the usage guide, but idk if thats too nitpicky

**Review by RehanSD — APPROVED** [review]
LGTM!

**Review by RehanSD — APPROVED** [review]
LGTM!

**Review by devin-petersohn — APPROVED** [review]
LGTM!

**Review by YarShev — APPROVED** [review]

**Pull request merged**